File: app/src/main/java/com/dumuzeyn/mp3player/SoundGroup.java
package com.dumuzeyn.mp3player;

import java.util.ArrayList;
import java.util.List;

final class SoundGroup {
    final String id;
    final String nameRussian;
    final String nameEnglish;
    final double[] centroid;
    final ArrayList<String> trackIds;

    SoundGroup(String id, String nameRussian, String nameEnglish, double[] centroid,
            List<String> trackIds) {
        this.id = id == null ? "" : id;
        this.nameRussian = nameRussian == null ? "" : nameRussian;
        this.nameEnglish = nameEnglish == null ? "" : nameEnglish;
        this.centroid = centroid == null ? new double[0] : centroid.clone();
        this.trackIds = new ArrayList<>(trackIds == null
                ? java.util.Collections.emptyList() : trackIds);
    }

    SoundGroup named(String russian, String english) {
        return new SoundGroup(id, russian, english, centroid, trackIds);
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/SoundGroup.kt
package com.dumuzeyn.mp3player

class SoundGroup(
    id: String?,
    nameRussian: String?,
    nameEnglish: String?,
    centroid: DoubleArray?,
    trackIds: List<String>?,
) {
    @JvmField val id: String = id.orEmpty()
    @JvmField val nameRussian: String = nameRussian.orEmpty()
    @JvmField val nameEnglish: String = nameEnglish.orEmpty()
    @JvmField val centroid: DoubleArray = centroid?.clone() ?: DoubleArray(0)
    @JvmField val trackIds: ArrayList<String> = ArrayList(trackIds.orEmpty())

    fun named(russian: String?, english: String?): SoundGroup =
        SoundGroup(id, russian, english, centroid, trackIds)
}
